### ISSUE TITLE
Increase HDD write speed

### ISSUE THREAD

**Issue by aramod** [madMAx43v3r/chia-gigahorse#46]
There is a suggestion how to increase the recording speed on HDD.
Make two working dir `-t` `-t1` for load balancing and alternate them in turn, this will make it possible to quickly copy on HDD
I have time PLOT 1.7 min. and I do not have time to copy it immediately at 8 HDD because -t is almost always busy 100%
For example:
`-t /mnt/ssd1/ -t1 /mnt/ssd2/ ....`
On ssd1 make 10 plots, then switch to ssd2 and also make 10 plots
This would give time to unload the ssd cache.

The first 5-10 sections are copied at a speed of 300-800mb/s, after 15-20plots the speed drops to 50-90mb/s. I don't know what it is, but adding -t1 solution would solve this problem as well.

**Comment by madMAx43v3r** [comment]
you are running out of trimmed space, the same would happen with adding another `-t1`, just later

**Comment by aramod** [comment]
Now I have to stop work for 10 minutes every 2-3 hours. During these 10 minutes, everything is copied to hdd very quickly without cuda_ploter working. After that, it again lasts for 2-3 hours

**Comment by madMAx43v3r** [comment]
Yeah it's an issue with the SSDs, trimming while reading / writing seems to be much slower...

**Comment by madMAx43v3r** [comment]
I can add support for multiple `-t`, that would solve other issues as well

**Comment by aramod** [comment]
It would be very wonderful! Thank you for your hard work!

**Comment by aramod** [comment]
And another question about VRAM.
I have 2 gpu 6gb each, I thought that it would work 6+6gb, but it turns out cuda_plotter only sees 6gb. Is it possible to do so that I don't have to use -S 3

**Comment by madMAx43v3r** [comment]
You need the same VRAM on each GPU, it's not added together.

**Comment by aramod** [comment]
And if he adds `fstrim ssd` after every 3-5 plots? This might help too or am I wrong?

**Comment by madMAx43v3r** [comment]
maybe yes, but I cannot do that in the plotter code

**Comment by aramod** [comment]
> И еще вопрос про VRAM. У меня 2 гпу по 6гб, думал что будет работать 6+6гб, а оказывается cuda_plotter видит только 6гб. Можно ли сделать так, чтобы мне не приходилось использовать`-S 3`

How can I start farming on gpu1 and run cuda_plotter gpu0? On the plotter `-r 1`, but on chia.bin?

**Comment by aramod** [comment]
I solved the problem with the temp disk by formatting btrfs on it.

```
sudo mkfs.btrfs -n 64k -d raid0 -m raid0 /dev/nvme0n1 /dev/nvme1n1 -f
mkdir /mnt/nvme
mount -o nodatacow,ssd,noatime,discard,nodatasum,nobarrier,autodefrag /dev/nvme0n1 /mnt/nvme
sudo chown -R user  /mnt/nvme
```
Added changes to /etc/fstab

Just in case, I changed the fstrim service `/usr/lib/systemd/system/fstrim.timer`
Changed text description from 'weekly' to 'hourly'.
Changed "weekly" to "hourly"
Changed accuracy from "1h" to "30s"

`/usr/lib/systemd/system/fstrim.service`
Changed ExecStart=/sbin/fstrim -av
```
sudo systemctl daemon-reload
sudo systemctl restart fstrim.timer
```
Now I do not have a queue and the recording speed does not drop after some time and is 250mb/s

**Comment by thecybo** [comment]
> I can add support for multiple `-t`, that would solve other issues as well

any timeline on this? I have 11 intermediary drives that worked very well with Bladebit and was confused why Gigahorse doesn't support it. Basically the (full, in-memory) plotter throws the finished plot on the first drive, then the next one on the second drive, etc, while I use plow to rsync them to their final destination.
With the current setup unless I want the plotter to pause from time to time I need to output the plot to the fastest NVME and from there move it to the intermediary drives (which is way slower with rsync than the plotter could output), but this just adds unnecessary writes and complexities to the flow.

**Comment by madMAx43v3r** [comment]
Multiple `-t` are now supported
